pkg/nats/proxy/compute_handler.go

```diff
@@ -126,21 +134,44 @@ func sendResponse[Response any](conn *nats.Conn, reply string, result *concurren
 	return conn.Publish(reply, resultData)
 }
 
-func processAndStream[Request, Response any](ctx context.Context, streamingClient *stream.Client, msg *nats.Msg,
+func processAndStream[Request, Response any](ctx context.Context, streamingClient *stream.ProducerClient, msg *nats.Msg,
 	f handlerWithResponse[Request, <-chan *concurrency.AsyncResult[Response]]) {
 	if msg.Reply == "" {
 		log.Ctx(ctx).Error().Msgf("streaming request on %s has no reply subject", msg.Subject)
 		return
 	}
+
 	writer := streamingClient.NewWriter(msg.Reply)
+	streamRequest := new(stream.Request)
+	err := json.Unmarshal(msg.Data, streamRequest)
+	if err != nil {
+		_ = writer.CloseWithCode(stream.CloseBadRequest,
```

Comment:
`CloseWithCode` returns an error in the event json marshaling fails, or if publishing the data to the subject fails. At a minimum we need to log the error from this method if one is returned. Please address this change here and elsewhere.